### PR TITLE
feature(integration): GA4 - add optional parameters for user reports

### PR DESCRIPTION
**Pull request by akashrpo**
## Description of the change

> add optional parameters in order to get user data/reports

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/ga4/transform.js [hunk 1]
```diff
@@ -242,11 +244,27 @@ const responseBuilder = (message, { Config }) => {
...
       message,
       ["properties"],
-      GA4_RESERVED_PARAMETER_EXCLUSION,
+      GA4_RESERVED_PARAMETER_EXCLUSION.concat(GA4_PARAMETERS_EXCLUSION),
       payload
     );
   }
 
+  // take optional engagement_time_msec, session_id parameters for reports
+  if (message.type === "identify" || message.type === "group") {
```
Comment (utsabc) [inline v0/destinations/ga4/transform.js:253]: Why can t we put this in the mapping json itself rather than writing extra code to set this?

**Comment by jjhuff** [comment]
Sweet, I think this'll address some of my GA4 problems.
FWIW, Segment seems to default `engagementTime` to `1` for `page` events because that seems to be needed to get decent stats. I can do this with a transformer, but it might be nice to do it here.

Also, any ideas on what to use for `sessionId`?

**Comment by akashrpo** [comment]
Hi @jjhuff, Session ID identifies the session that an event came from. These are automatically collected via gtag.js However, the identifier is not included automatically in events from Measurement Protocol. Reference - https://support.google.com/analytics/answer/9191807?hl=en#zippy=

**Review by utsabc — APPROVED** [review]

File: v0/destinations/ga4/transform.js [hunk 2]
```diff
@@ -129,6 +130,15 @@ const responseBuilder = (message, { Config }) => {
         payload
       );
     }
+
```
Comment (ItsSudip) [inline v0/destinations/ga4/transform.js:133]: Don't we have to add `GA4_PARAMETERS_EXCLUSION` here in the exclusionFileds?

File: v0/destinations/ga4/transform.js [hunk 3]
```diff
@@ -219,6 +247,15 @@ const responseBuilder = (message, { Config }) => {
       getGA4ExclusionList(mappingConfig[ConfigCategory.GROUP.name]),
       payload
     );
+
```
Comment (ItsSudip) [inline v0/destinations/ga4/transform.js:250]: Same here.

File: v0/destinations/ga4/transform.js (hunk 2 above)
Comment (akashrpo) [reply v0/destinations/ga4/transform.js:133]: getGA4ExclusionList() is handling it

File: v0/destinations/ga4/transform.js (hunk 3 above)
Comment (akashrpo) [reply v0/destinations/ga4/transform.js:250]: getGA4ExclusionList() is handling it

**Review by ItsSudip — APPROVED** [review]

**Pull request merged**